### ISSUE TITLE
Incorrect name resolution for JSDoc template tag constraints in typedef

### ISSUE THREAD

**Issue by rbuckton** [microsoft/TypeScript#45480]
# Bug Report

### 🔎 Search Terms

jsdoc name resolve lookup symbol template constraint typedef

### 🕗 Version & Regression Information

- This has been an issue since we added support for `@template` in JSDoc

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?target=99&jsx=0&filetype=js#code/PTAEAEGMAsFNINYCkDOAuUAXATgV1gLABQIEAhgDYUD2A7qhjvsaeCjgJaSaN6FEsAVIOKhBETLAC2ABwplJoACqjx4SbPmKA3glgBPagDNlAX1ABpVRP0zYAE1gmAYtWqrgxIA)

### 💻 Code

```js
/**
 * @template T
 * @template {keyof T} K
 * @typedef Foo
 */
```

### 🙁 Actual behavior

Reports `` Cannot find name `T` ``.

### 🙂 Expected behavior

Does not report an error.

### Notes

Name resolution does not look at the symbol table of the `@typedef` tag, so when looking for `T` in the code below, we end up looking at `T` for the function declaration further down in this example:

```js
/**
 * @template T
 * @template {keyof T} K
 * @typedef Foo
 */

/**
 * @template T
 * @param {T} a
 */
function bar(a) {
}
```

A "go-to-definition" on the `T` in `keyof T` jumps to the `T` declared for `bar`, not the `T` in the typedef for `Foo`.

**Comment by awerlogus** [comment]
Duplicate of https://github.com/microsoft/TypeScript/issues/43403
